src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/StringSearchValues.cs

```diff
@@ -0,0 +1,365 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using System.Runtime.Intrinsics.X86;
+using System.Text;
+using static System.Buffers.StringSearchValuesHelper;
+
+namespace System.Buffers
+{
+    internal static class StringSearchValues
+    {
+        private static readonly SearchValues<char> s_asciiLetters =
+            SearchValues.Create("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz");
+
+        public static SearchValues<string> Create(ReadOnlySpan<string> values, bool ignoreCase)
```

Comment:
I know you've measured throughput of the resulting instance.  Have you measured Create overhead in typical cases?

Reply:
`Create` called 1000 times with most frequent words in Sherlock (1 value per `SearchValues`):
`CreateIgnoreCase` already has values in their uppercase equivalent.

|                         Method |     Mean |    Gen0 | Allocated |
|------------------------------- |---------:|--------:|----------:|
|                         Create | 126.8 us | 26.8555 | 265.28 KB |
|               CreateIgnoreCase | 143.4 us | 26.8555 | 265.28 KB |
| CreateIgnoreCase_FromLowercase | 144.0 us | 30.5176 | 299.94 KB |

`Create` called 1 time with 1000 most frequent words in Sherlock (AhoCorasick path for lots of values):

|                         Method |     Mean |   Error | Allocated |
|------------------------------- |---------:|--------:|----------:|
|                         Create | 262.0 us | 2.67 us | 189.81 KB |
|               CreateIgnoreCase | 233.1 us | 1.15 us | 184.97 KB |
| CreateIgnoreCase_FromLowercase | 238.5 us | 1.15 us | 219.63 KB |

`Create` for `["Sherlock", "Holmes", "Watson", "Irene", "Adler", "John", "Baker"]` (Teddy + RabinKarp):

|                         Method |     Mean | Allocated |
|------------------------------- |---------:|----------:|
|                         Create | 1.388 us |   1.96 KB |
|               CreateIgnoreCase | 1.447 us |   1.96 KB |
| CreateIgnoreCase_FromLowercase | 1.535 us |    2.2 KB |

`Create` called 1 time with 32 most frequent words in Sherlock (bucketized Teddy + RabinKarp):

|                         Method |     Mean | Allocated |
|------------------------------- |---------:|----------:|
|                         Create | 5.287 us |   6.72 KB |
|               CreateIgnoreCase | 5.435 us |   8.63 KB |
| CreateIgnoreCase_FromLowercase | 5.645 us |   9.61 KB |

Reply:
Ok, thanks.  Certainly not something we'd want to be doing a lot of, but reasonable given what we expect it to save on subsequent searches.

Reply:
> `Create` called 1000 times with most frequent words in Sherlock (1 value per `SearchValues`):

Avoiding the `HashSet` and `string[]` allocations for the single-value case as well, these are down to ~100 ns per `Create` with 1 allocation (2 if we have to reallocate the string).

|                         Method |      Mean | Allocated |
|------------------------------- |----------:|----------:|
|                         Create |  78.76 us |  54.98 KB |
|               CreateIgnoreCase |  94.43 us |  54.69 KB |
| CreateIgnoreCase_FromLowercase | 106.16 us |   93.7 KB |